### PR TITLE
insights: Add 4 new backend dashboard pings

### PR DESCRIPTION
**Pull request by CristinaBirkel**
Closes https://github.com/sourcegraph/sourcegraph/issues/28374

## Description

This PR adds 4 new backend-calculated insights pings, for dashboards.

- **TotalOrgs**: This ping exists at the top level, outside of `codeInsightsUsage`
- **TotalOrgsWithDashboard**
- **TotalDashboardCount**
- **InsightsPerDashboard** (stats)

[Schema PR](https://github.com/sourcegraph/analytics/pull/364) which needs to be merged in before this one.
[Docs PR](https://github.com/sourcegraph/sourcegraph/pull/32092)

## Test plan

I tested this locally by examining the output of the pings at `site-admin/pings`. Here is an example of the 4 new pings:

```
"totalOrgs": 6,        
"codeInsightsUsage": {
        "InsightsPerDashboard": {
            "Avg": 3.3703704,
            "Max": 10,
            "Min": 1,
            "Median": 3,
            "StdDev": 2.6039636
        },
        "TotalDashboardCount": 41,
        "TotalOrgsWithDashboard": 4,
        ... (rest of our existing pings)
}
```

### CODE REVIEW

File: enterprise/internal/insights/background/pings/insights_ping_aggregators.go [hunk 1]
```diff
@@ -112,6 +113,44 @@ func (e *InsightsPingEmitter) GetOrgVisibleInsightCounts(ctx context.Context) (_
...
+	if err != nil {
+		return types.InsightsPerDashboardPing{}, err
+	}
+	defer func() { err = rows.Close() }()
+
+	var insightsPerDashboardStats types.InsightsPerDashboardPing
+	rows.Next()
+	if err := rows.Scan(
+		&insightsPerDashboardStats.Avg,
+		&insightsPerDashboardStats.Min,
```
Comment (coury-clark) [inline enterprise/internal/insights/background/pings/insights_ping_aggregators.go:143]: Looks like the min/max order are swapped in this `Scan` versus the query below

**Review by coury-clark — APPROVED** [review]

File: enterprise/internal/insights/background/pings/insights_ping_aggregators.go (hunk 1 above)
Comment (CristinaBirkel) [reply enterprise/internal/insights/background/pings/insights_ping_aggregators.go:143]: You're right, great catch! Ah, I even posted it in the Test Plan section like that.. will fix!

**Comment by coury-clark** [comment]
Reminder - we need to update the schema before merging

**Review by Joelkw — APPROVED** [review]
Thanks @CristinaBirkel ! I actually think none of these should be prefixed with `Weekly` now that I see we're just implementing them as simple counts – my bad in the RFC naming. The `weekly` preface goes to pings that reset each week, which these don't (they just snapshot the current count). It looks like our equivalent insight charts count ping is just called "InsightTotalCounts" without the weekly. 

You could move org count out of the insights object if it's easy enough, because I know some others may want it. 

Also, please do update the pings doc (either as part of this PR or in a fast-follow PR) so it records this addition!

File: internal/types/types.go [hunk 2]
```diff
@@ -1222,6 +1225,14 @@ type InsightTotalCounts struct {
 	ViewSeriesCounts []InsightViewSeriesCountPing
 }
 
+type InsightsPerDashboardPing struct {
+	Avg    float32
+	Max    int
+	Min    int
+	StdDev float32
+	Median int
```
Comment (ebrodymoore) [inline internal/types/types.go:1233]: Couldn't be median also be a float if there's an odd number of dashboards?

Comment (CristinaBirkel) [reply internal/types/types.go]: Yes, thank you! Great catch. Just fixed that (and on the schema change.)

**Pull request merged**